### ISSUE TITLE
Feature request: Data sources

### ISSUE THREAD

**Issue by jsimpso** [terraform-lxd/terraform-provider-lxd#506]
It would be great if we could use the provider to reference existing resources as [data sources](https://developer.hashicorp.com/terraform/language/data-sources), e.g. referencing a network by name to have another resource act on its IPv4 range:

```terraform
data "lxd_network" "my_net" {
  name     = "my-net"
  project  = "my-project"
}

resource "example" "example" {
  ip_range = data.lxd_network.my_net.ipv4
}
```

**Comment by stew3254** [comment]
I agree. This would make it much easier for Terraform to manage things that already exist. Especially something like the `default` profile created with every project. One of my pet peeves is that I can't change the default project on an image to use my new default, or to simply manage this one. To work around that I've been using the `external` provider and wrote a script to copy the data over from the `def` profile I make, but there is no reason this can't be included as a data source in Terraform.

**Comment by MusicDin** [comment]
Basic datasources, such as project, profile, image, network, storage pool, and instance are planned for the upcoming release.

> Especially something like the default profile created with every project.

Just to add on this one, the default profile can be managed for non-default projects. Of course this is not a replacement for a datasource, but just to point it out :)

```hcl
resource "lxd_project" "myproject" {
  name = "myproject"
  config = {
    "features.profiles": true
  }
}

resource "lxd_profile" "default" {
  name        = "default"
  project     = lxd_project.myproject.name
  description = "Modified default profile description"
}
```

**Comment by stew3254** [comment]
I did not know that. It's been a while since I tried. Thanks for letting me know, that actually fixes something I'm doing!
